checkov/common/util/http_utils.py

```diff
@@ -204,36 +206,54 @@ def request_wrapper(
 
 
 async def aiohttp_client_session_wrapper(
-        url: str,
+        method: str,
+        url: StrOrURL,
         headers: dict[str, Any],
-        payload: dict[str, Any]
-) -> int:
+        payload: dict[str, Any] | None = None,
+) -> tuple[int, Any]:
     request_max_tries = int(os.getenv('REQUEST_MAX_TRIES', 3))
     sleep_between_request_tries = float(os.getenv('SLEEP_BETWEEN_REQUEST_TRIES', 1))
 
+    # 1. Read proxy URL (may include user:pass for authentication)
+    proxy_url = os.environ.get("HTTPS_PROXY", os.environ.get("HTTP_PROXY"))
+    proxy_auth = None
+    if proxy_url:
+        parsed_proxy_url = urlparse(proxy_url)
+        if parsed_proxy_url.username and parsed_proxy_url.password:
+            proxy_auth = aiohttp.BasicAuth(login=parsed_proxy_url.username, password=parsed_proxy_url.password)
+    # 2. Read path to custom certificate bundle
+    ca_bundle_path = os.getenv('BC_CA_BUNDLE')
+    ssl_context = None
+    if ca_bundle_path:
+        logger.info(f"Loading custom CA bundle from: {ca_bundle_path}")
+        # Create a new SSL context
+        ssl_context = ssl.create_default_context(cafile=ca_bundle_path)
+    connector = aiohttp.TCPConnector(resolver=aiohttp.AsyncResolver(), ssl_context=ssl_context)
+
     # adding retry mechanism for avoiding the next repeated unexpected issues:
     # 1. Gateway Timeout from the server
     # 2. ClientOSError
-    async with aiohttp.ClientSession(connector=aiohttp.TCPConnector(resolver=aiohttp.AsyncResolver())) as session:
+    async with aiohttp.ClientSession(connector=connector) as session:
         for i in range(request_max_tries):
             logging.info(
                 f"[http_utils](aiohttp_client_session_wrapper) reporting attempt {i + 1} out of {request_max_tries}")
             try:
-                async with session.post(
-                        url=url, headers=headers, json=payload
+                async with session.request(
+                        method=method, url=url, headers=headers, json=payload, proxy=proxy_url, proxy_auth=proxy_auth
                 ) as response:
                     content = await response.text()
+                    response_json = await response.json()
                 if response.ok:
                     logging.info(f"[http_utils](aiohttp_client_session_wrapper) - done successfully to url: \'{url}\'")
-                    return 0
+                    return 0, response_json
                 elif i != request_max_tries - 1:
                     await asyncio.sleep(sleep_between_request_tries * (i + 1))
                     continue
                 else:
                     logging.error(f"[http_utils](aiohttp_client_session_wrapper) - Failed to send report to "
                                   f"url \'{url}\'")
                     logging.error(f"Status code: {response.status}, Reason: {response.reason}, Content: {content}")
-                    return 1
+                    return 1, None
```

Comment:
why do you return both return code and json? isn't it better to just return the response itself, and the relvent caller will decide what to do with it?
If you do want to do it like that, let's create an object for it to be more readable (even a `namedTuple` can be good)